src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Unix.cs

```diff
@@ -154,6 +154,41 @@ internal static unsafe IntPtr AllocBSTR(int length)
             return s;
         }
 
+        internal static unsafe IntPtr SysAllocStringByteLen(byte[]? str, uint length)
```

Comment:
It may look better to make this symmetric with `AllocBSTR`, just take the length and leave it up to the caller to copy the content in.

Reply:
Renamed to `AllocBSTRByteLen()` and removed copy implementation.